game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java

```diff
@@ -948,22 +957,17 @@ public IBattle getPendingBattle(final GUID guid) {
     if (guid == null) {
       return null;
     }
-    for (final IBattle battle : pendingBattles) {
-      if (guid.equals(battle.getBattleId())) {
-        return battle;
-      }
-    }
-    return null;
+
+    return pendingBattles.stream()
+        .filter(b -> b.getBattleId().equals(guid))
+        .findAny()
+        .orElse(null);
```

Comment:
This sounds like it would make sense to use a HashMap to look up battles.

Reply:
Happily the length of `pendingBattles` will never be enough where we need a hashmap for the O(1) lookup, performance should not matter here.

In this case not unreasonable to use a hashmap, though looks like there is not a very good SRP around `pendingBattles`, we also do a lookup by territory as well in another method.